traversal/iterator/GraphIterator.java

```diff
@@ -52,7 +55,8 @@
     private final List<Identifier.Variable.Name> filter;
     private final Map<Identifier, ResourceIterator<? extends Vertex<?, ?>>> iterators;
     private final Map<Identifier, Vertex<?, ?>> answer;
-    private final Map<Identifier.Variable, Set<ThingVertex>> scoped;
+    private final Map<Identifier.Variable, Set<ThingVertex>> scopedRoles;
+    private final Map<Identifier.Variable, PriorityQueue<Integer>> scopedEdgeOrders;
```

Comment:
I think this should be wrapped in a dedicated class now, which provides one API that manages both datasets synchronously under-the-hood.